benchmark/_interfaces.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2015 ClusterHQ Inc.  See LICENSE file for details.
+"""
+Operations for the control service benchmarks.
+"""
+
+from zope.interface import Interface
+
+
+class IScenario(Interface):
+    """
+    A scenario in which an operation occurs.
+    """
+
+    def __init__(self, clock, control_service):
+        """
+        Provide the scenario with a clock and control service.
+        """
+
+    def start(self):
+        """
+        :return: A Deferred that fires when the desired scenario is
+            established (e.g. that a certain load is being applied).
+        """
+
+    def maintained(self):
+        """
+        :return: A Deferred that fires with an errback if the desired
+            scenario fails to hold between being established and being
+            stopped.  This Deferred never fires with a callback.
+        """
```

Comment:
I did think of calling it `collapsed` and making it use a callback if the scenario collapses. In that case, errBack would indicate a problem in monitoring the status.  Since both a collapsing scenario and a problem monitoring whether the scenario has collapsed are problematic for the benchmark, the calling code would probably be expected to use `addBoth` to handle. (That's why I didn't bother in the end). But maybe it makes more sense?